http-clients/src/main/java/com/palantir/remoting/http/UserAgentInterceptor.java

```diff
@@ -40,8 +41,8 @@ public static UserAgentInterceptor of(String userAgent) {
 
     @Override
     public void apply(RequestTemplate template) {
-        if (!template.headers().containsKey("User-Agent")) {
-            template.header("User-Agent", userAgent);
+        if (!HeaderAccessUtils.caseInsensitiveContains(template.headers(), HttpHeaders.USER_AGENT)) {
```

Comment:
+1

On Fri, Jul 8, 2016 at 11:14 AM Mark Elliot <notifications@github.com>
wrote:

> In
> http-clients/src/main/java/com/palantir/remoting/http/UserAgentInterceptor.java
> <https://github.com/palantir/http-remoting/pull/156#discussion_r70117093>:
>
> > @@ -40,8 +41,8 @@ public static UserAgentInterceptor of(String userAgent) {
> >
> >      @Override
> >      public void apply(RequestTemplate template) {
> > -        if (!template.headers().containsKey("User-Agent")) {
> > -            template.header("User-Agent", userAgent);
> > +        if (!HeaderAccessUtils.caseInsensitiveContains(template.headers(), HttpHeaders.USER_AGENT)) {
>
> to expand on that, the test we could do for caseInsensitiveContains
> should do the loop but not the map creation.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/156/files/492ffaccad904a7857dc4e50b068c96c12b98b4d#r70117093>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGOdwfJWun6_UdDa-m4mYXriVZvQwPy6ks5qTpOMgaJpZM4JHoer>
> .
>